src/DynExpManager/Instruments/NP_Conex_CC.h

```diff
@@ -0,0 +1,344 @@
+		};
+
+		class MoveToHomeTask final : public DynExp::TaskBase
+		{
+		public:
+			MoveToHomeTask(CallbackType CallbackFunc) noexcept : TaskBase(CallbackFunc) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+		};
+
+		class MoveToHomeExecutionTask final : public DynExp::TaskBase
+		{
+		public:
+			MoveToHomeExecutionTask(CallbackType CallbackFunc, std::chrono::system_clock::time_point DeferUntil = {}) noexcept
+				: TaskBase(CallbackFunc, DeferUntil) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+		};
+
+		class MoveAbsoluteTask final : public DynExp::TaskBase
+		{
+		public:
+			MoveAbsoluteTask(PositionerStageData::PositionType Position, CallbackType CallbackFunc) noexcept
+				: TaskBase(CallbackFunc), Position(Position) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+
+			const PositionerStageData::PositionType Position;
+		};
+
+		class MoveAbsoluteExecutionTask final : public DynExp::TaskBase
+		{
+		public:
+			MoveAbsoluteExecutionTask(PositionerStageData::PositionType Position, CallbackType CallbackFunc, std::chrono::system_clock::time_point DeferUntil = {}) noexcept
+				: TaskBase(CallbackFunc, DeferUntil), Position(Position) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+
+			const PositionerStageData::PositionType Position;
+
+		};
+
+		class MoveRelativeTask final : public DynExp::TaskBase
+		{
+		public:
+			MoveRelativeTask(PositionerStageData::PositionType Position, CallbackType CallbackFunc) noexcept
+				: TaskBase(CallbackFunc), Position(Position) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+
+			const PositionerStageData::PositionType Position;
+		};
+
+		class MoveRelativeExecutionTask final : public DynExp::TaskBase
+		{
+		public:
+			MoveRelativeExecutionTask(PositionerStageData::PositionType Position, CallbackType CallbackFunc, std::chrono::system_clock::time_point DeferUntil = {}) noexcept
+				: TaskBase(CallbackFunc, DeferUntil), Position(Position) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+
+			const PositionerStageData::PositionType Position;
+
+		};
+
+		class StopMotionTask final : public DynExp::TaskBase
+		{
+		public:
+			StopMotionTask(CallbackType CallbackFunc = nullptr, std::chrono::system_clock::time_point DeferUntil = {}) noexcept
+				: TaskBase(CallbackFunc, DeferUntil) {}
+
+		private:
+			virtual DynExp::TaskResultType RunChild(DynExp::InstrumentInstance& Instance) override;
+		};
+	}
+
+	class NP_Conex_CCStageData : public PositionerStageData
+	{
+		friend class NP_Conex_CC_Tasks::UpdateTask;
+
+	public:
+		struct Conex_CCStatusType
+		{
+			constexpr void Set(uint8_t byteCode) noexcept { ByteCode = byteCode; }
```

Comment:
My convention is

`constexpr void Set(uint8_t ByteCode) noexcept { this->ByteCode = ByteCode; }`